include/SDL3/SDL_init.h

```diff
@@ -182,6 +188,106 @@ extern SDL_DECLSPEC SDL_InitFlags SDLCALL SDL_WasInit(SDL_InitFlags flags);
+extern SDL_DECLSPEC int SDLCALL SDL_SetAppMetadata(const char *appname, const char *appversion, const char *appidentifier);
+
+
+/**
+ * Specify metadata about your app through a set of properties.
+ *
+ * You can optionally provide metadata about your app to SDL. This is not
+ * required, but strongly encouraged.
+ *
+ * There are several locations where SDL can make use of metadata (an "About"
+ * box in the macOS menu bar, the name of the app can be shown on some audio
+ * mixers, etc). Any piece of metadata can be left out, if a specific detail
+ * doesn't make sense for the app.
+ *
+ * This function should be called as early as possible, before SDL_Init.
+ * Multiple calls to this function are allowed, but various state might not
+ * change once it has been set up with a previous call to this function.
+ *
+ * Any previously set values not included in `props` will be removed.
+ * Passing a 0 for `props` removes all previous metadata.
+ *
+ * These are the supported properties:
+ *
+ * - `SDL_PROP_APP_METADATA_NAME_STRING`: The human-readable name of
+ *   the application, like "My Game 2: Bad Guy's Revenge!"
+ * - SDL_PROP_APP_METADATA_VERSION_STRING`: The version of the app that
+ *   is running; there are no rules on format, so "1.0.3beta2" and
+ *   "April 22nd, 2024" and a git hash are all valid options.
+ * - `SDL_PROP_APP_METADATA_IDENTIFIER_STRING`: A unique string that
+ *   identifies this app. This must be in reverse-domain format, like
+ *   "com.example.mygame2". This string is used by desktop compositors to
+ *   identify and group windows together, as well as match applications
+ *   with associated desktop settings and icons.
+ * - SDL_PROP_APP_METADATA_CREATOR_STRING`: The human-readable name
+ *   of the creator/developer/maker of this app, like "MojoWorkshop, LLC"
+ * - SDL_PROP_APP_METADATA_COPYRIGHT_STRING`: The human-readable copyright
+ *   notice, like "Copyright (c) 2024 MojoWorkshop, LLC" or whatnot. Keep
+ *   this to one line, don't paste a copy of a whole software license in
+ *   here.
+ * - SDL_PROP_APP_METADATA_URL_STRING`: A URL to the app on the web. Maybe
+ *   a product page, or a storefront, or even a GitHub repository, for
+ *   user's further information.
+ * - SDL_PROP_APP_METADATA_TYPE_STRING`: The type of application this is.
+ *   Currently this string can be "game" for a video game, "mediaplayer" for
+ *   a media player, or generically "application" if nothing else applies.
+ *   Future versions of SDL might add new types.
+ *
+ * \param props the properties to use.
+ * \returns 0 on success or a negative error code on failure; call
+ *          SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \threadsafety Do not call this function from two threads at once.
+ *
+ * \sa SDL_SetAppMetadata
+ */
+extern SDL_DECLSPEC int SDLCALL SDL_SetAppMetadataWithProperties(SDL_PropertiesID props);
+
+#define SDL_PROP_APP_METADATA_NAME_STRING         "name"
```

Comment:
```suggestion
#define SDL_PROP_APP_METADATA_NAME_STRING         "SDL.app.metadata.name"
```
?

Reply:
We can do this, but SDL_CreateWindowWithProperties went with bare words, so I did, too. They aren't in a global namespace. Whichever way we go, we should be consistent with other functions.

Reply:
In that case the "_CREATE_" properties are distinct from the final properties. Maybe that's silly and we shouldn't have distinct names for them?

Reply:
I went with this change in 164928aff9b82ae3a43940b0dbe41307439fbaba. Seems fancier.  :)